aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/core/util/PluginReflectionUtility.java

```diff
@@ -188,23 +189,101 @@ public static PluginReflectionUtility fromCodeScope(List<String> elements, Strin
     /**
      * Gets generic list of handler instances invoked from all available derivatives of specified handler {@code Class}.
      * Each is supplied with a reference to {@link PluginRuntimeContext} as required
+     *
      * @param handlerClass {@code Class} object
-     * @param <T> Expected handler type
+     * @param <T>          Expected handler type
      * @return {@link List<T>} of instances
      */
     private <T> List<T> getHandlers(Class<? extends T> handlerClass) {
-        return reflections.getSubTypesOf(handlerClass).stream()
+        List<T> list = reflections.getSubTypesOf(handlerClass).stream()
                 .map(PluginReflectionUtility::getHandlerInstance)
                 .filter(Objects::nonNull)
                 .sorted(Comparator.comparing(handler -> handler.getClass().getCanonicalName())) // to provide stable handlers sequence between runs
                 .collect(Collectors.toList());
+        Graph classGraph = new Graph(list);
+        for (T item : list) {
+            Class<?> after = item.getClass().getDeclaredAnnotation(Handles.class).after();
+            Class<?> before = item.getClass().getDeclaredAnnotation(Handles.class).before();
+            classGraph.addEdge(item.getClass(), before);
+            classGraph.addEdge(after, item.getClass());
+        }
+        List<T> sortedList = (List<T>) classGraph.topologicalSort();
+        List<T> finalList = new ArrayList<>();
+        int i = 0;
+        int k = 0;
+        while (finalList.size() != list.size()) {
+            if (list.get(i).getClass().equals(sortedList.get(k))) {
+                finalList.add(list.get(i));
+                k++;
+                i = 0;
+            } else {
+                i++;
+            }
+        }
+        return list;
+    }
+
+    /**
+     * Implements topological sorting
+     */
+    public class Graph {
+        private HashMap<Class<?>, ArrayList> edges;
+        private List nodes;
+        private List nodeVisited;
+        private ArrayList edgeList;
+
+        public Graph(List vertices) {
+            nodes = vertices;
+            edges = new HashMap();
+            nodeVisited = new ArrayList();
+        }
+
+        public void addEdge(Class<?> x, Class<?> y) {
+            if (!edges.containsKey(x)) {
+                edgeList = new ArrayList();
+            } else {
+                edgeList = edges.get(x);
+            }
+            edgeList.add(y);
+            edges.put(x, edgeList);
+        }
+
+        public <T> List<Class<? extends T>> topologicalSort() {
+            Stack stack = new Stack();
```

Comment:
Please introduce typization. Then you'' nee no type coersion such as at line 260